Library/Homebrew/keg_relocate.rb

```diff
@@ -70,12 +72,22 @@ def relocate_text_files(old_prefix, new_prefix, old_cellar, new_cellar)
 
   def change_dylib_id(id, file)
     puts "Changing dylib ID of #{file}\n  from #{file.dylib_id}\n    to #{id}" if ARGV.debug?
-    install_name_tool("-id", id, file)
+
+    if ENV["HOMEBREW_RUBY_MACHO"]
+      MachO::Tools.change_dylib_id(file, id)
+    else
+      install_name_tool("-id", id, file)
+    end
   end
```

Comment:
Was just thinking it might be nice to use `Module`s here and `include` them based on the environment variable.